compose/cli/main.py

```diff
@@ -655,20 +655,19 @@ def remove_container(force=False):
         if options['--rm']:
             project.client.remove_container(container.id, force=True)
 
-    def force_shutdown(signal, frame):
-        project.client.kill(container.id)
-        remove_container(force=True)
-        sys.exit(2)
-
-    def shutdown(signal, frame):
-        set_signal_handler(force_shutdown)
-        project.client.stop(container.id)
-        remove_container()
-        sys.exit(1)
+    signals.set_signal_handler_to_shutdown()
+    try:
+        dockerpty.start(project.client, container.id, interactive=not options['-T'])
+        exit_code = container.wait()
```

Comment:
It looks like there are some race conditions around here, for example if ForceShutdownException occurs at this line.